perf-tool/src/monitor.cpp

```diff
@@ -214,4 +184,15 @@ JNIEXPORT void JNICALL MonitorContendedEntered(jvmtiEnv *jvmtiEnv, JNIEnv *env,
     }
     
     sendToServer(j.dump());
+
+    /* Also call the callback */
+    EventConfig::CallbackIDs callbackIDs = monitorConfig.getCallBackIDs(env);
+    if (callbackIDs.cachedCallbackClass && callbackIDs.cachedCallbackMethodId)
+    {
+        jstring jsonEntry = env->NewStringUTF(j.dump().c_str());
+        if (jsonEntry)
+        { 
+            env->CallStaticVoidMethod(callbackIDs.cachedCallbackClass, callbackIDs.cachedCallbackMethodId, jsonEntry);
```

Comment:
You probably need to check for exceptions here too.

Reply:
Fixed